src/Bicep.Core/Syntax/TestDeclarationSyntax.cs

```diff
@@ -53,9 +53,31 @@ this.Value switch
                 // blocked by assert in the constructor
                 _ => throw new NotImplementedException($"Unexpected type of test value '{this.Value.GetType().Name}'.")
             };
+        
+        public ObjectSyntax? TryGetParameters(ObjectPropertySyntax property)=>
+            property.Value switch
+            {
+                ObjectSyntax @object => @object,
+                SkippedTriviaSyntax => null,
+
+                // blocked by assert in the constructor
+                _ => throw new NotImplementedException($"Unexpected type of params value '{this.Value.GetType().Name}'.")
+            };
+        
 
         public ObjectSyntax GetBody() =>
             this.TryGetBody() ?? throw new InvalidOperationException($"A valid test body is not available on this test due to errors. Use {nameof(TryGetBody)}() instead.");
 
+        public ObjectSyntax GetParameters(){
```

Comment:
```suggestion
        public ObjectSyntax? TryGetParameters(){
```

Not every module defines parameters, so not every value assigned in a `test` statement will have a `params` property. Rather than just returning the full assignment body in that case, you should return `null` (since there are no parameters defined).